### ISSUE TITLE
Nickname "made the parser go crazy, striping"

### ISSUE THREAD

**Issue by chy168** [emesene/emesene#103]
When I login, I got some error mesg:

```
this nick: '({)青い魚です(}) (／‵′)／~ ╧╧ ...
[$small]ooxx@hotmail.com[$/small]
[$small](busy) - 不喜歡下雨天...[$/small]' made the parser go crazy, striping
```

I think that is the problem of Japanese and Chinese words.

I found the error mesg is shown in "gui/gtkui/Renderers.py" line 140.
I want to contribute but new at PyGTK and Python.

Any idea to solve this issue ?

**Comment by EibrielInv** [comment]
Maybe you must paste the nicks on pasteall.org, or leave four spaces before each line, so the webpage dont format it, example:
    Hello world!

**Comment by chy168** [comment]
OK, thanks for your suggestion!!

**Comment by c10ud** [comment]
i think ({) (}) are triggering some weird stuff in the parser (like a pixbuf load or smth...lol), it's not because of the japanese/chinese character..can you confirm that?

**Comment by chy168** [comment]
Yes, I confirm the crazy from ({) or (}) like you said.
I reproduce the crazy parser by changing nickname to "({)TEST".

I trace it to "Parser.py", there is a "FilterData" class, it return a wrong split data list to "data" in line 456.
I want to fix this problem, but I cannot understand the FilterData class (maybe GObject ?).

I've also read emesene 1.6's code, seems the same parsing.
The nickname works well at 1.6.

thank you :p

**Comment by c10ud** [comment]
maybe it's easier than it seems: i think it's a missing icon (or, two). specifically hugleft and hugright!

**Comment by c10ud** [comment]
actually, they're not missing, but they're not loading...

**Comment by c10ud** [comment]
Fixed "hugs" crashing the parser. We'd better NOT run git blame on that lines, i think. Closed by 7fcd845d1d0218779aacc6f06336192078aa0d6b
